Sources/MuxUploadSDK/InternalUtilities/Reporting/Reporter.swift

```diff
@@ -33,83 +40,248 @@ class Reporter: NSObject {
 
-    private func generateRequest(
-        url: URL,
-        httpBody: Data
-    ) -> URLRequest {
-        let request = NSMutableURLRequest(url: url,
-                                          cachePolicy: .useProtocolCachePolicy,
-                                          timeoutInterval: 10.0)
-        request.httpMethod = "POST"
-        request.setValue("application/json", forHTTPHeaderField: "Accept")
-        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
-        request.httpBody = httpBody
-
-        return request as URLRequest
+    func reportUploadInputStandardizationFailure(
+        errorDescription: String,
+        inputDuration: Double,
+        inputSize: UInt64,
+        nonStandardInputReasons: [String],
+        options: UploadOptions,
+        standardizationEndTime: Date,
+        standardizationStartTime: Date,
+        uploadCanceled: Bool,
+        uploadURL: URL
+    ) {
+        guard !options.eventTracking.optedOut else {
+            return
+        }
+
+        let data = InputStandardizationFailedEvent.Data(
+            appName: Bundle.main.appName,
+            appVersion: Bundle.main.appVersion,
+            deviceModel: device.model,
+            errorDescription: errorDescription,
+            inputDuration: inputDuration,
+            inputSize: inputSize,
+            maximumResolution: options.inputStandardization.maximumResolution.description,
+            nonStandardInputReasons: nonStandardInputReasons,
+            platformName: device.systemName,
+            platformVersion: device.systemVersion,
+            regionCode: regionCode,
+            sdkVersion: Version.versionString,
+            standardizationStartTime: standardizationStartTime,
+            standardizationEndTime: standardizationEndTime,
+            uploadCanceled: uploadCanceled,
+            uploadURL: uploadURL
+        )
+
+        let event = InputStandardizationFailedEvent(
+            sessionID: sessionID,
+            version: Version.versionString,
+            data: data
+        )
+
+        send(
+            event: event,
+            url: url
+        )
     }
 }
 
 // TODO: Implement as a separate object so the URLSession
 // can become non-optional, which removes a bunch of edge cases
 extension Reporter: URLSessionDelegate, URLSessionTaskDelegate {
     func urlSession(_ session: URLSession, task: URLSessionTask, willPerformHTTPRedirection response: HTTPURLResponse, newRequest request: URLRequest, completionHandler: @escaping (URLRequest?) -> Swift.Void) {
-        if(self.pendingUploadEvent != nil) {
-            if let redirectUrl = request.url, let httpBody = try? serializePendingEvent() {
-                let request = self.generateRequest(
-                    url: redirectUrl,
-                    httpBody: httpBody
-                )
-                completionHandler(request)
+        if let pendingEvent = pendingEvents[ObjectIdentifier(task)], let redirectURL = request.url {
```

Comment:
`Reporter` can now manage multiple in-flight requests and can now be shared, this is a very simplistic implementation, not thread-safe, but will work reasonably outside of weird edge cases.